tests/test_server.py

```diff
@@ -1266,22 +1289,34 @@ async def test_resource_deploy_done(server, client, environment, agent, caplog,
     assert resource_action["action"] == const.ResourceAction.deploy
     assert resource_action["started"] is not None
     assert resource_action["finished"] is not None
-    assert resource_action["messages"] == [
+
+    expected_timestamp: str
+    if opt.server_tz_aware_timestamps.get():
+        if endpoint_to_use == "resource_deploy_done":
+            expected_timestamp = now.astimezone().isoformat(timespec="microseconds")
+        else:
+            expected_timestamp = now.astimezone(timezone.utc).isoformat(timespec="microseconds")
+    else:
+        expected_timestamp = now.astimezone(timezone.utc).replace(tzinfo=None).isoformat()
+
+    parse_timestamp(expected_timestamp)
```

Comment:
This feels a bit hackish

Reply:
Agreed, I had hoped you could make it simpler but I'd overlooked the fact that the result actually contains the strings. Still, I prefer this over what you had before. I think the next step would be to validate the result with pydantic so you get an actual `datetime.datetime` object, and then you don't need to convert `expected_timestamp` to a specific timezone or to `str`. But that adds complexity in another place so I don't know if it's worth it. No strong preference from me.